src/Akka.Persistence.Redis/Snapshot/RedisSnapshotStore.cs

```diff
@@ -99,4 +102,13 @@ private SelectedSnapshot PersistentFromBytes(byte[] bytes)
 
         private string GetSnapshotKey(string persistenceId) => $"{_settings.KeyPrefix}snapshot:{persistenceId}";
     }
+
+    internal static class SnapshotMetadataExtensions
+    {
+        public static bool Matches(this SnapshotSelectionCriteria criteria, SnapshotMetadata metadata)
```

Comment:
We should make this method - public
https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence/SnapshotProtocol.cs#L539

And to create a spec which ensures that a plugin respects all snapshot criteria